### PR TITLE
A personal TODO list

### PR DESCRIPTION
**Pull request by j2l**
I bring something to help user track his stuff using a todo list in DM only (just to be more intimate than exposed bot commands, but feel free to change this).
Discovered Klasa a few weeks ago so please be kind to my nooblishness.
I PR now to give back to the community and also learn more about pieces, klasa-vs and the klasa ecosystem in general.

For the command to run, once need to add to your main server file:
```
Client.defaultUserSchema
  .add("TODOs", "any", {array: true, default: [], configurable: true});
```
I don't know how to do it with piece, so I added it as a comment. Feel free to show me how and I'll implement.

I heavily lean on klasa-tags (that I K.I.S.S.) and help from the community, so thanks!, and not really my copyright on this one (which is different from a license BTW).

If it's a piece of junk, feel free to explain me how I could improve it technically.

### CODE REVIEW

File: commands/Tools/todo.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	}
+
+	async remove(message, [TODO]) {
+		const filtered = message.author.settings.TODOs.filter(([name]) => name !== TODO);
+		await message.author.settings.update('TODOs', filtered, { action: 'overwrite' });
+		return message.send(`Removed \`${TODO}\` TODO`);
+	}
+
+	list(message) {
+		return message.send(`TODOs for this author are: ${message.author.settings.TODOs.map(key => `\n${key}`)}`);
```
Comment (kyranet) [inline commands/Tools/todo.js]:
```suggestion
		return message.send(`List of TODOs for this user: \`${message.author.settings.TODOs.join('`, `')}\``);
```

And possibly check if TODOs has at least one element.

File: commands/Tools/todo.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  .add("TODOs", "any", {array: true, default: [], configurable: true});
+*/
+
+const { Command } = require('klasa');
+
+module.exports = class extends Command {
+
+	constructor(...args) {
+		super(...args, {
+			enabled: true,
```
Comment (kyranet) [inline commands/Tools/todo.js]: Please nuke all optional properties that are defaults.

File: commands/Tools/todo.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
+/*
+To your main server file, add:
+Client.defaultUserSchema
+  .add("TODOs", "any", {array: true, default: [], configurable: true});
```
Comment (kyranet) [inline commands/Tools/todo.js]:
```suggestion
  .add('TODOs', 'any', { array: true });
```

When `array` is set to true, `default` becomes an empty array by default, and when type is `any`, `configurable` is always set to false.

File: commands/Tools/todo.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+/*
+To your main server file, add:
```
Comment (kyranet) [inline commands/Tools/todo.js]:
```suggestion
This piece requires a `TODOs` key to work, add in your main file:
```

File: commands/Tools/todo.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+			autoAliases: true,
+			bucket: 1,
+			cooldown: 0,
+			promptLimit: 0,
+			promptTime: 30000,
+			guarded: false,
+			permissionLevel: 0,
+			description: 'add|remove|list user\'s TODOs through DM',
+			extendedHelp: 'No extended help available.',
+			usage: '<add|remove|list> (TODO:string) [content:...string]',
```
Comment (kyranet) [inline commands/Tools/todo.js]:
```suggestion
			usage: '<add|remove|list:default> (TODO:string) [content:...string]',
```

So when you do `@bot todo`, it shows you the list. It's more intuitive than not defaulting.

File: commands/Tools/todo.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+			usageDelim: ' ',
+			quotedStringSupport: false,
+			subcommands: true
+		});
+		//
```
Comment (kyranet) [inline commands/Tools/todo.js]: What's this empty comment?

Comment (j2l) [reply commands/Tools/todo.js]: I don't remember

File: commands/Tools/todo.js (hunk 2 above)
Comment (kyranet) [reply commands/Tools/todo.js]: This property is `true` by default
```suggestion
```

File: commands/Tools/todo.js (hunk 6 above)
Comment (kyranet) [reply commands/Tools/todo.js]:
```suggestion
```

**Comment by j2l** [comment]
Sorry, I don't have time anymore to follow up this PR.
Please commit or close it.

**Pull request merged**